src/main/java/org/codehaus/mojo/versions/DisplayPluginUpdatesMojo.java

```diff
@@ -292,7 +292,8 @@ else if ( event.isEndElement() )
     {
         // we want only those parts of pluginManagement that are defined in this project
         Map<String, String> pluginManagement = new HashMap<>();
-        try
+
+        if ( model.getBuild().getPluginManagement() != null )
         {
             for ( Plugin plugin : model.getBuild().getPluginManagement().getPlugins() )
```

Comment:
You can still get NullPointers here, if `model.getBuild().getPluginManagement().getPlugins()` returns null.

As long as you removed the try/catch statement, I think there should be a nullity verfiy before the for loop, or put back the try/catch

Reply:
I checked every getter if it could return `null` or not. A call to `getPlugins()` can never return `null`. Furthermore we would have a lot of `NullPointerExceptions` during the integration tests.